codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -191,11 +205,24 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
         }
     }
 
-    private void onStreamActive(int streamId) {
-        ChannelFuture future = createStreamChannel(ctx, streamId, streamHandler);
-        Http2StreamChannel childChannel = (Http2StreamChannel) future.channel();
-        Http2StreamChannel oldChannel = childChannels.put(streamId, childChannel);
-        assert oldChannel == null;
+    private void onStreamActive(int streamId, Http2HeadersFrame headersFrame) {
+        final Http2StreamChannel childChannel;
+        if (isOutbound(streamId)) {
+            if (!(headersFrame instanceof ChannelCarryingHeadersFrame)) {
+                throw new IllegalArgumentException("needs to be wrapped");
+            }
+            childChannel = ((ChannelCarryingHeadersFrame) headersFrame).channel();
+            childChannel.streamId(streamId);
+        } else {
+            Channel parentChannel = ctx.channel();
+            EventLoopGroup group = streamGroup != null ? streamGroup : parentChannel.eventLoop();
+            ChannelFuture future =
+                    createStreamChannel(parentChannel, group, streamHandler, streamOpts, streamAttrs, streamId);
+            childChannel = (Http2StreamChannel) future.channel();
+        }
+
+        Http2StreamChannel existing = childChannels.put(streamId, childChannel);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "existing". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1854)